tests/src/unit-optional.cpp

```diff
@@ -0,0 +1,159 @@
+#include "doctest_compatibility.h"
+
+#include <nlohmann/json.hpp>
+
+#if (defined(__cplusplus) && __cplusplus >= 201703L) || (defined(_HAS_CXX17) && _HAS_CXX17 == 1) // fix for issue #464
+    #define JSON_HAS_CPP_17
+    #define JSON_HAS_CPP_14
+#elif (defined(__cplusplus) && __cplusplus >= 201402L) || (defined(_HAS_CXX14) && _HAS_CXX14 == 1)
+    #define JSON_HAS_CPP_14
+#endif
```

Comment:
This shouldn't be needed. `JSON_HAS_CPP_*` macros from the library are now available to the unit tests.